pyfr/__main__.py

```diff
@@ -201,83 +200,82 @@ def process_partition_info(args):
+        # Check it does not already exist unless --force is given
+        if pname in mesh['partitionings'] and not args.force:
+            raise ValueError('Partitioning already exists; use -f to replace')
 
-    # Create the partitioner
-    if args.partitioner:
-        part = get_partitioner(args.partitioner, pwts, ewts, opts=opts)
-    else:
-        for name in sorted(cls.name for cls in subclasses(BasePartitioner)):
-            try:
-                part = get_partitioner(name, pwts, ewts)
-                break
-            except OSError:
-                pass
-        else:
-            raise RuntimeError('No partitioners available')
+        # Path to store the partitioning in the mesh
+        ppath = f'partitionings/{pname}'
 
-    # Partition the mesh
-    pinfo = part.partition(mesh, args.progress)
-    (partitioning, pregions), (neighbours, nregions) = pinfo
+        # Partitioner-specific options
+        opts = dict(s.split(':', 1) for s in args.popts)
 
-    # Write out the new partitioning
-    with args.progress.start('Write partitioning'):
-        if ppath in mesh:
-            mesh[f'{ppath}/eles'][:] = partitioning
-            del mesh[f'{ppath}/neighbours']
+        # Create the partitioner
+        if args.partitioner:
+            part = get_partitioner(args.partitioner, pwts, ewts, opts=opts)
         else:
-            mesh[f'{ppath}/eles'] = partitioning
-
-        mesh[f'{ppath}/neighbours'] = neighbours
-        mesh[f'{ppath}/eles'].attrs['regions'] = pregions
-        mesh[f'{ppath}/neighbours'].attrs['regions'] = nregions
+            for name in sorted(cls.name for cls in subclasses(BasePartitioner)):
+                try:
+                    part = get_partitioner(name, pwts, ewts)
+                    break
+                except OSError:
+                    pass
+            else:
+                raise RuntimeError('No partitioners available')
+
+        # Partition the mesh
+        pinfo = part.partition(mesh, args.progress)
+        (partitioning, pregions), (neighbours, nregions) = pinfo
+
+        # Write out the new partitioning
+        with args.progress.start('Write partitioning'):
+            if ppath in mesh:
+                mesh[f'{ppath}/eles'][:] = partitioning
+                del mesh[f'{ppath}/neighbours']
+            else:
+                mesh[f'{ppath}/eles'] = partitioning
+
+            mesh[f'{ppath}/neighbours'] = neighbours
+            mesh[f'{ppath}/eles'].attrs['regions'] = pregions
+            mesh[f'{ppath}/neighbours'].attrs['regions'] = nregions
 
 
 def process_partition_remove(args):
     with h5py.File(args.mesh, 'r+') as mesh:
         mparts = mesh['partitionings']
 
-        if args.name not in mparts:
-            raise ValueError(f'Partitioning {args.name} does not exist')
+    if args.name not in mparts:
+        raise ValueError(f'Partitioning {args.name} does not exist')
 
-        del mparts[args.name]
+    del mparts[args.name]
```

Comment:
This will error; we are trying to delete something from a closed file (since it is outside the `with` block).